contracts/interchain-token/tests/test.rs

```diff
@@ -4,55 +4,515 @@ extern crate std;
+    assert_eq!(token.balance(&user2), 0);
+
+    assert_invoke_auth_ok!(user1, token.try_burn(&user1, &amount2));
+    assert_eq!(
+        env.auths(),
+        std::vec![(
+            user1.clone(),
+            AuthorizedInvocation {
+                function: AuthorizedFunction::Contract((
+                    token.address.clone(),
+                    symbol_short!("burn"),
+                    (&user1, amount2).into_val(&env),
+                )),
+                sub_invocations: std::vec![]
+            }
+        )]
+    );
+
+    assert_eq!(token.balance(&user1), 0);
+    assert_eq!(token.balance(&user2), 0);
+}
+
+#[test]
+#[should_panic(expected = "HostError: Error(Context, InvalidAction)")]
+fn decimal_is_over_max() {
+    let env = Env::default();
+    let owner = Address::generate(&env);
+    let minter = Address::generate(&env);
+    let interchain_token_service = Address::generate(&env);
+    let token_id: BytesN<32> = BytesN::<32>::random(&env);
+    let token_meta_data = TokenMetadata {
+        decimal: (u32::from(u8::MAX) + 1),
+        name: "name".into_val(&env),
+        symbol: "symbol".into_val(&env),
+    };
+
+    env.register(
+        InterchainToken,
         (
-            Symbol::new(&env, "ownership_transferred"),
             owner,
-            new_owner.clone(),
+            minter,
+            &interchain_token_service,
+            &token_id,
+            token_meta_data,
+        ),
+    );
+}
+
+#[test]
+#[should_panic(expected = "HostError: Error(Context, InvalidAction)")]
+fn token_name_is_empty() {
+    let env = Env::default();
+    let owner = Address::generate(&env);
+    let minter = Address::generate(&env);
+    let interchain_token_service = Address::generate(&env);
+    let token_id: BytesN<32> = BytesN::<32>::random(&env);
+    let token_meta_data = TokenMetadata {
+        decimal: 1,
+        name: "".into_val(&env),
+        symbol: "symbol".into_val(&env),
+    };
+
+    env.register(
+        InterchainToken,
+        (
+            owner,
+            minter,
+            &interchain_token_service,
+            &token_id,
+            token_meta_data,
+        ),
+    );
+}
+
+#[test]
+#[should_panic(expected = "HostError: Error(Context, InvalidAction)")]
+fn token_symbol_is_empty() {
+    let env = Env::default();
+    let owner = Address::generate(&env);
+    let minter = Address::generate(&env);
+    let interchain_token_service = Address::generate(&env);
+    let token_id: BytesN<32> = BytesN::<32>::random(&env);
+    let token_meta_data = TokenMetadata {
+        decimal: 1,
+        name: "name".into_val(&env),
+        symbol: "".into_val(&env),
+    };
+
+    env.register(
+        InterchainToken,
+        (
+            owner,
+            minter,
+            &interchain_token_service,
+            &token_id,
+            token_meta_data,
+        ),
+    );
+}
+
+#[test]
+#[should_panic(expected = "HostError: Error(Context, InvalidAction)")]
+fn token_id_is_empty() {
```

Comment:
this shouldn't be a test anymore. this is checking symbol is empty

Reply:
make sense. removed